### PR TITLE
Creationpane layout

### PR DESCRIPTION
**Pull request by Markvanderuit**

### CODE REVIEW

**Comment by bart-de-jonge** [comment]
You call this branch css-basics, but htis is no basics anymore. Please keep PR a LOT smaller than this.

File: src/main/java/gui/misc/BlurHelper.java [hunk 1]
```diff
@@ -1,46 +1,60 @@
...
+ * Also has support for moving elements.
+ * Is kind of extremely buggy. May have horrible performance impact.
  */
 public class BlurHelper {
 
-    private WritableImage writableImage;    // Image used for writing snapshots to.
-    private SnapshotParameters parameters; // Parameters used for writing snapshots (resolution etc)
-    private Bounds bounds; // Bounds of node used for snapshot parameters (x, y, width, height)
-
+    /**
```
Comment (bart-de-jonge) [inline src/main/java/gui/misc/BlurHelper.java]: Javadoc usually belongs to a method or specific variable. This comment is about multiple variable, so make it a general line or block comment.

File: src/main/java/gui/misc/BlurHelper.java [hunk 2]
```diff
@@ -1,46 +1,60 @@
...
-    private Node node; // Node behind which we're blurring
-
     @Getter @Setter
     private boolean hideNode; // Whether or not node is included in blurring process
-
     @Getter @Setter
     private Point2D offset; // Pixel offset if we wish to move blur slightly
 
     /**
+     * Object and scene variables.
```
Comment (bart-de-jonge) [inline src/main/java/gui/misc/BlurHelper.java:43]: Same, change javadoc to general line or block comment

Comment (Markvanderuit) [inline src/main/java/gui/misc/BlurHelper.java:43]: Done. Running through all other classes now and changing it there as well.

File: src/main/java/gui/misc/BlurHelper.java [hunk 3]
```diff
@@ -124,4 +142,24 @@ public void processBlurUsingBounds() {
         imageView.setImage(writableImage);
     }
 
+    /**
+     * Bind this blurhelper to scrollchanges in a watchable scrollpane.
+     * @param watchable the scrollpane to watch.
+     */
+    public void watchScrolling(ScrollPane watchable) {
+        ChangeListener<Number> listener = new ChangeListener<Number>() {
```
Comment (bart-de-jonge) [inline src/main/java/gui/misc/BlurHelper.java]: maybe extract this into a seperate method and use a lambda, also use a more descriptive name

File: src/main/java/gui/misc/TransitionHelper.java [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+import javafx.animation.Timeline;
+import javafx.beans.property.DoubleProperty;
+import javafx.beans.property.Property;
+import javafx.event.EventHandler;
+import javafx.scene.Node;
+import javafx.scene.input.MouseEvent;
+import javafx.util.Duration;
+import lombok.Getter;
+
+
```
Comment (bart-de-jonge) [inline src/main/java/gui/misc/TransitionHelper.java]: whitespace

File: src/main/java/gui/misc/TransitionHelper.java [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+    /**
+     * Constructor of class.
+     * @param node the node for which we want transitions.
+     */
+    public TransitionHelper(Node node) {
+        this.node = node;
+        transitionStores = new ArrayList<Double>();
+    }
+
+    /**
```
Comment (bart-de-jonge) [inline src/main/java/gui/misc/TransitionHelper.java]: change to regular comment

File: src/main/java/gui/misc/BlurHelper.java (hunk 3 above)
Comment (Markvanderuit) [inline src/main/java/gui/misc/BlurHelper.java]: That method is horribly broken and won't be used again. I just removed it.

File: src/main/java/gui/misc/TransitionHelper.java [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+        Timeline t2 = new Timeline();
+
+        // Create event handlers from x to y, and from y to x.
+        EventHandler<MouseEvent> mouseInHandler = createHandlerTowardsDouble(property, t1, t2, ms,
+                v, false, interpolator);
+        EventHandler<MouseEvent> mouseOutHandler = createHandlerTowardsDouble(property, t1, t2, ms,
+                v, true, interpolator);
+
+        // Bind event handlers on mouse enter and exit for this Node.
+        node.addEventHandler(MouseEvent.MOUSE_PRESSED, mouseInHandler);
```
Comment (bart-de-jonge) [inline src/main/java/gui/misc/TransitionHelper.java]: I think inlining the handler initialisation

File: src/main/java/gui/misc/TransitionHelper.java [hunk 7]
```diff
@@ -0,0 +1,322 @@
...
+     * with a specified transition half-time, over a specified property.
+     * @param property property being modified. (for example, color property)
+     * @param ms time in milliseconds for transition.
+     * @param x value before transition.
+     * @param y value after transition.
+     * @param <T> generic type.
+     */
+    public <T> void addMouseClickTransition(Property<T> property, int ms,
+                                            T x, T y) {
+        addMouseClickTransition(property, ms, x, y, Interpolator.LINEAR);
```
Comment (bart-de-jonge) [inline src/main/java/gui/misc/TransitionHelper.java]: Is this one-line method necessary?

File: src/main/java/gui/misc/TransitionHelper.java [hunk 8]
```diff
@@ -0,0 +1,322 @@
...
+    /**
+     * Add a transition event, on mouse enter and exit, between a specified double
+     * value and a specified double offset.
+     * @param property the property being modified.
+     * @param ms the time in milliseconds for transition.
+     * @param v the offset value.
+     */
+    public void addMouseOverTransition(DoubleProperty property, int ms,
+                                       double v) {
+        addMouseOverTransition(property, ms, v, Interpolator.LINEAR);
```
Comment (bart-de-jonge) [inline src/main/java/gui/misc/TransitionHelper.java]: Same story, a method body of one line, is this necessary?

File: src/main/java/gui/misc/TransitionHelper.java [hunk 9]
```diff
@@ -0,0 +1,322 @@
...
+     * with a specified transition half-time, over a specified property
+     * @param property property being modified. (for example, color property)
+     * @param ms time in milliseconds for transition.
+     * @param x value before transition.
+     * @param y value after transition.
+     * @param <T> generic type.
+     */
+    public <T> void addMouseOverTransition(Property<T> property, int ms,
+                                       T x, T y) {
+        addMouseOverTransition(property, ms, x, y, Interpolator.LINEAR);
```
Comment (bart-de-jonge) [inline src/main/java/gui/misc/TransitionHelper.java]: one-line method

File: src/main/java/gui/misc/TransitionHelper.java [hunk 10]
```diff
@@ -0,0 +1,322 @@
...
+        EventHandler<MouseEvent> mouseOutHandler = createHandlerBetweenGenerics(property, ms,
+                y, x, interpolator);
+
+        // Bind event handlers on mouse enter and exit for this Node.
+        node.addEventHandler(javafx.scene.input.MouseEvent.MOUSE_ENTERED, mouseInHandler);
+        node.addEventHandler(javafx.scene.input.MouseEvent.MOUSE_EXITED, mouseOutHandler);
+    }
+
+    /**
+     * Event handler creation functions below.
```
Comment (bart-de-jonge) [inline src/main/java/gui/misc/TransitionHelper.java:150]: change to regular comment

File: src/main/java/gui/misc/TransitionHelper.java (hunk 7 above)
Comment (Ezlo123) [inline src/main/java/gui/misc/TransitionHelper.java]: Unused. Removed.

File: src/main/java/gui/modal/CreationModalView.java [hunk 11]
```diff
@@ -1,33 +1,69 @@
...
 import javafx.scene.layout.HBox;
 import javafx.scene.layout.VBox;
 import javafx.scene.paint.Color;
-import javafx.scene.text.Text;
 import lombok.Getter;
 import lombok.Setter;
 
-import java.util.ArrayList;
-import java.util.List;
+
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/CreationModalView.java]: whitespace

File: src/main/java/gui/modal/CreationModalView.java [hunk 12]
```diff
@@ -1,33 +1,69 @@
...
 /**
  * Class responsible for displaying a modal view for the creation of shots.
  * @author alex
  */
 public class CreationModalView extends ModalView {
 
-    private static final int width = 600;
-    private static final int height = 600;
+    /**
+     * Tweakable styling variables.
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/CreationModalView.java:31]: change to regular comment

File: src/main/java/gui/modal/CreationModalView.java [hunk 13]
```diff
@@ -1,33 +1,69 @@
...
+    private int buttonFontSize = 16;
+
+    // color of the "active" element of a checkbox
+    private Point3D checkboxColor = new Point3D(250, 120, 50);
+
+    // misc
+    private Point3D titlelabelColor = new Point3D(255, 255, 255);
+
+    /**
+     * Misc variables.
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/CreationModalView.java]: change to regualar comment

File: src/main/java/gui/modal/CreationModalView.java [hunk 14]
```diff
@@ -81,73 +129,174 @@ public CreationModalView(RootPane rootPane, int numberOfCamerasInTimeline,
...
+        this.buttonPane.setSpacing(20.0);
+        this.buttonPane.setAlignment(Pos.CENTER_LEFT);
+        this.buttonPane.setPrefHeight(10000);
+        this.buttonPane.setStyle(bottomStyle);
+        this.buttonPane.setPadding(new Insets(0, 0, 0, 20));
+        this.viewPane.getChildren().add(buttonPane);
+
+        // Add cancel button
+        cancelButton = new StyledButton("Cancel");
+        cancelButton.setOnMouseReleased(e -> {
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/CreationModalView.java]: please add this listener in the controller (if now controller is present for the creationmodalview please make one and move the other createmodelview functionality from toolviewcontroller to the new controller)

File: src/main/java/gui/modal/CreationModalView.java [hunk 15]
```diff
@@ -81,73 +129,174 @@ public CreationModalView(RootPane rootPane, int numberOfCamerasInTimeline,
...
+        );
+        cancelButton.setPrefWidth(buttonWidth);
+        cancelButton.setPrefHeight(buttonHeight);
+        cancelButton.setFontSize(buttonFontSize);
+        cancelButton.setButtonColor(createButtonColor);
+        cancelButton.setAlignment(Pos.CENTER);
+
+        // Add creation button
+        creationButton = new StyledButton("Create");
+        creationButton.setOnMouseReleased(this::createShot);
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/CreationModalView.java]: Add this handler in the controller

File: src/main/java/gui/modal/CreationModalView.java [hunk 16]
```diff
@@ -81,73 +129,174 @@ public CreationModalView(RootPane rootPane, int numberOfCamerasInTimeline,
...
+     * Initialize all textfields, add them to a left-central VBox.
      */
-    private void initNameDescriptionFields() {
-        // Add the textfields for shot's name & description
+    private void initTextFields() {
+        VBox content = new VBox(10);
+        content.setAlignment(Pos.CENTER_LEFT);
+        content.setMinWidth(350.0);
+        content.setPrefWidth(10000);
+        content.setPrefHeight(10000);
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/CreationModalView.java]: please make constant of these

File: src/main/java/gui/modal/CreationModalView.java [hunk 17]
```diff
@@ -81,73 +129,174 @@ public CreationModalView(RootPane rootPane, int numberOfCamerasInTimeline,
...
+
+    /**
+     * Initializes pane with buttons at bottom.
+     */
+    private void initButtons() {
+        // setup button pane
+        this.buttonPane = new HBox();
+        this.buttonPane.setSpacing(20.0);
+        this.buttonPane.setAlignment(Pos.CENTER_LEFT);
+        this.buttonPane.setPrefHeight(10000);
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/CreationModalView.java]: please use constants

File: src/main/java/gui/modal/CreationModalView.java [hunk 18]
```diff
@@ -81,73 +129,174 @@ public CreationModalView(RootPane rootPane, int numberOfCamerasInTimeline,
...
     /**
-     * Initialize the shot name and description fields.
+     * Initialize title label.
+     */
+    private void initTitleLabel() {
+        titleLabel = new Label("Add a new shot...");
+        titleLabel.setStyle(topStyle);
+        titleLabel.setAlignment(Pos.CENTER_LEFT);
+        titleLabel.setPadding(new Insets(0, 0, 0, 20));
+        titleLabel.setPrefWidth(10000);
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/CreationModalView.java]: please use constants

File: src/main/java/gui/modal/CreationModalView.java [hunk 19]
```diff
@@ -81,73 +129,174 @@ public CreationModalView(RootPane rootPane, int numberOfCamerasInTimeline,
...
     /**
-     * Initialize the checkboxes with labels for each camera.
+     * Initialize the checkboxes with labels for each camera, in a flowpane.
      */
     private void initCamCheckBoxes() {
+        // Create new FlowPane to hold the checkboxes.
+        this.checkboxPane = new FlowPane();
+        this.checkboxPane.setHgap(20.0);
+        this.checkboxPane.setVgap(20.0);
+        this.checkboxPane.setMinWidth(300.0);
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/CreationModalView.java]: please use constants

File: src/main/java/gui/styling/StyledCheckbox.java [hunk 20]
```diff
@@ -0,0 +1,249 @@
...
+     * @param r component of color.
+     * @param g component of color.
+     * @param b component of color.
+     */
+    public void setMarkColor(int r, int g, int b) {
+        Platform.runLater(new Runnable() {
+            @Override
+            public void run() {
+                mark.setStyle(mark.getStyle().concat("-fx-background-color: rgb("
+                    + r + ","
```
Comment (bart-de-jonge) [inline src/main/java/gui/styling/StyledCheckbox.java]: is it necessary this is on three lines, doesn't add too much to readability

File: src/main/java/gui/styling/StyledCheckbox.java [hunk 21]
```diff
@@ -0,0 +1,249 @@
...
+     * @param g component of color.
+     * @param b component of color.
+     */
+    public void setBoxColor(int r, int g, int b) {
+        Platform.runLater(new Runnable() {
+            @Override
+            public void run() {
+                box.setStyle(box.getStyle().concat("-fx-background-color: rgb("
+                        + r + ","
+                        + g + ","
```
Comment (bart-de-jonge) [inline src/main/java/gui/styling/StyledCheckbox.java]: Check comment above

File: src/main/java/gui/styling/StyledTextfield.java [hunk 22]
```diff
@@ -0,0 +1,113 @@
...
+    /**
+     * Simple function to set color of textfield, rgb style, 0-255.
+     * @param r red component of color.
+     * @param g green component of color.
+     * @param b blue component of color.
+     */
+    public void setTextfieldColor(int r, int g, int b) {
+        setStyle(getStyle().concat("-fx-background-color: rgb("
+                + r + ","
+                + g + ","
```
Comment (bart-de-jonge) [inline src/main/java/gui/styling/StyledTextfield.java]: See previous comment

File: src/main/java/gui/styling/StyledTextfield.java [hunk 23]
```diff
@@ -0,0 +1,113 @@
...
+    /**
+     * Simple function to set color of text, rgb style, 0-255.
+     * @param r red component of color.
+     * @param g green component of color.
+     * @param b blue component of color.
+     */
+    public void setTextColor(int r, int g, int b) {
+        setStyle(getStyle().concat("-fx-text-fill: rgb("
+                + r + ","
+                + g + ","
```
Comment (bart-de-jonge) [inline src/main/java/gui/styling/StyledTextfield.java]: See previous comment

File: src/main/java/gui/modal/CreationModalView.java (hunk 14 above)
Comment (Markvanderuit) [inline src/main/java/gui/modal/CreationModalView.java]: Not implemented. Rest of team disagrees on this decision.

Comment (bart-de-jonge) [inline src/main/java/gui/modal/CreationModalView.java]: The team disagrees on putting the validation in the controller, the button pressed listener is agreed to be moved to a controller

Comment (Markvanderuit) [inline src/main/java/gui/modal/CreationModalView.java]: The team doesn't agree on anything. I'm not touching that part of the code anymore, someone else can move it and get shot.

**Pull request closed without merging**